.rubocop_todo.yml

```diff
@@ -0,0 +1,49 @@
+# This configuration was generated by
+# `rubocop --auto-gen-config`
+# on 2016-03-27 01:01:12 +0100 using RuboCop version 0.38.0.
+# The point is for the user to remove these configuration records
+# one by one as the offenses are removed from the code base.
+# Note that changes in the inspected code, or installation of new
+# versions of RuboCop, may require this file to be generated again.
+
+# I don't have a preference.
+Lint/Loop:
+  Enabled: false
+
+# I'd rather use context_dependent, even if I'm not even sure we should enforce a style.
+# SupportedStyles: braces, no_braces, context_dependent
+Style/BracesAroundHashParameters:
+  Enabled: false
+  EnforcedStyle: context_dependent
+
+# Offense count: 39
+Style/Documentation:
+  Enabled: false
+  Exclude:
+    - 'spec/**/*'
+    - 'test/**/*'
+
+# I was a big fan of leading, but after I had some weird compile issues, I reverted
+# to trailing. Any proposal?
+# SupportedStyles: leading, trailing
+Style/DotPosition:
+  Enabled: false
+
+# It doesn't seem to work as expected.
+Style/IfUnlessModifier:
+  Enabled: false
+  MaxLineLength: 80
+
+# raise Foo, "bar" is cool, but when we have multiple args it collapses.
+Style/RaiseArgs:
+  Enabled: false
+
+# I don't have a preference.
+Style/SpecialGlobalVars:
+  EnforcedStyle: use_perl_names
+
+# We generally use double quotes, sometimes single quotes.
+# Should we enforce it at code level?
+# SupportedStyles: single_quotes, double_quotes
```

Comment:
@jacegu yes, that's my (personal) guideline, mostly used for consistency across the existing codes. However, such standard it's hard to enforce with a tool like rubocop.

That's why I vote to either disable it, or switch to double quote.

Double quote are preferred for me to single because:

- Most languages consistently represent strings with single quote
- They handle interpolation and other evaluations, hence we don't have to switch between one format and another when special evaluations are required